frontend/src/components/costTable/CostTable.vue

```diff
@@ -0,0 +1,255 @@
+    ellipsis: {
+      tooltip: true,
+    },
+    sorter: {
+      // sort by the case-insensitive alphabetical order
+      compare: (row1: DataRow, row2: DataRow) => {
+        const a = row1.region.toLocaleLowerCase();
+        const b = row2.region.toLocaleLowerCase();
+        const len = a.length > b.length ? b.length : a.length;
+        const stringComp = a.localeCompare(b);
+        if (stringComp !== 0) {
+          return stringComp;
+        }
+
+        // if tow region are identical, sort them with id
+        return row1.id - row2.id;
+      },
+      multiple: 1,
+    },
+    rowSpan: (rowData: DataRow) => {
+      return rowData.childCnt;
+    },
+  },
+  {
+    title: "CPU",
+    key: "cpu",
+    align: "right",
+    sorter: {
+      compare: (row1: DataRow, row2: DataRow) => row1.cpu - row2.cpu,
+      multiple: 2,
+    },
+    rowSpan: (rowData: DataRow) => {
+      return rowData.childCnt;
+    },
+    ellipsis: {
+      tooltip: false,
+    },
+    render(row: DataRow) {
+      let shortenProcessor = "";
+      if (window.innerWidth > 1080) {
+        shortenProcessor = row.processor.split(" ").slice(0, 2).join(" ");
+      } else if (window.innerWidth > 800) {
+        shortenProcessor = row.processor.split(" ")[0];
+      }
+
+      // if the screen is too short, hide the processor
+      let render: any;
+      if (window.innerWidth > 800) {
+        render = h("div", {}, [
+          h(
+            NTag,
+            {
+              round: true,
+              class: "inline mr-2 ",
+              type: row.processor === "" ? "warning" : "info",
+              size: "small",
+              bordered: false,
+            },
+            {
+              default: () => (row.processor === "" ? "N/A" : shortenProcessor),
+            }
+          ),
+          h("div", { class: "inline-block text-right w-6" }, row.cpu),
+        ]);
+      } else {
+        // if the screen is too short, hide the processor
+        render = row.cpu;
+      }
+
+      return h(
+        NTooltip,
+        {},
+        {
+          default: () => (row.processor === "" ? "N/A" : row.processor),
+          trigger: () => render,
+        }
+      );
+    },
+  },
+  {
+    title: "Memory",
+    key: "memory",
+    align: "right",
+    defaultSortOrder: false,
+    sorter: {
+      compare: (row1: DataRow, row2: DataRow) =>
+        Number(row1.memory) - Number(row2.memory),
+      multiple: 2,
+    },
+    rowSpan: (rowData: DataRow) => {
+      return rowData.childCnt;
+    },
+  },
+  {
+    title: "Pricing",
+    key: "pricing",
+    align: "center",
+    children: getPricingContent(),
+    ellipsis: {
+      tooltip: true,
+    },
+  },
+]);
+
+const props = defineProps({
+  dataRow: {
+    type: Array as PropType<DataRow[]>,
+    default: () => [],
+  },
+  isLoading: { type: Boolean, default: () => false },
+  showEngineType: { type: Boolean, default: () => false },
+  showLeaseLength: { type: Boolean, default: () => false },
```

Comment:
`default: false` is okay, when property's type is a primitive type (number, boolean, string)